nautobot_device_lifecycle_mgmt/filters.py

```diff
@@ -164,24 +171,20 @@ class Meta:
 
         fields = "__all__"
 
-    def search(self, queryset, name, value):  # pylint: disable=unused-argument
-        """Perform the filtered search."""
-        if not value.strip():
-            return queryset
-
-        qs_filter = (
-            Q(version__icontains=value)
-            | Q(alias__icontains=value)
-            | Q(release_date__icontains=value)
-            | Q(end_of_support__icontains=value)
-        )
-        return queryset.filter(qs_filter)
-
 
 class SoftwareImageLCMFilterSet(NautobotFilterSet):
     """Filter for SoftwareImageLCM."""
 
-    q = django_filters.CharFilter(method="search", label="Search")
+    q = SearchFilter(
+        filter_predicates={
+            "image_file_name": {"lookup_expr": "icontains", "preprocessor": str.strip},
+            "software__version": {"lookup_expr": "icontains", "preprocessor": str.strip},
+            "device_types__model": {"lookup_expr": "icontains", "preprocessor": str.strip},
+            "inventory_items__id": {"lookup_expr": "icontains", "preprocessor": str.strip},
+            "object_tags__name": {"lookup_expr": "icontains", "preprocessor": str.strip},
+            "devices__name": {"lookup_expr": "icontains", "preprocessor": str.strip},
```

Comment:
I'm not seeing a `devices` field on this model:

```
django.core.exceptions.FieldError: Cannot resolve keyword 'devices' into field. Choices are: _custom_field_data, associated_contacts, associated_object_metadata, created, default_image, destination_for_associations, device_types, download_url, hashing_algorithm, id, image_file_checksum, image_file_name, inventory_items, last_updated, migrated_to_core_model, migrated_to_core_model_flag, migrated_to_core_model_id, object_tags, software, software_id, source_for_associations, static_group_association_set, tagged_items, tags
```

This model is going away so it's probably not worth spending too much time on it.